cmd/segment-reaper/detect_test.go

```diff
@@ -0,0 +1,635 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"bytes"
+	"encoding/csv"
+	"fmt"
+	"math"
+	"math/bits"
+	"math/rand"
+	"testing"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/skyrings/skyring-common/tools/uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"storj.io/storj/pkg/pb"
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/private/testcontext"
+	"storj.io/storj/satellite/metainfo"
+	"storj.io/storj/storage/teststore"
+)
+
+func TestObserver(t *testing.T) {
+	t.Run("processSegment", func(t *testing.T) {
```

Comment:
If that's clear for you, it's also better for me, however, I would name those functions `TestObserver_processSegment` and `TestObserver_analyzeProject`.

Reply:
names are ok for me. Moving to separate functions will also reduce indentation a little bit.

Reply:
I like that, and it's my preference but I found people preferring to scope since Go released the `testing.Run` method. I prefer to use them but at the same time not having many nested scopes, and what you suggested it's what I would have done in a personal project.

Reply:
Done